src/reformat_libyuv.c

```diff
@@ -50,6 +56,126 @@ unsigned int avifLibYUVVersion(void)
 #pragma clang diagnostic pop
 #endif
 
+static avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * rgb);
+
+avifResult avifImageRGBToYUVLibYUV(avifImage * image, const avifRGBImage * rgb)
+{
+    if ((image->depth == 8) && (rgb->depth == 8)) {
+        return avifImageRGBToYUVLibYUV8bpc(image, rgb);
+    }
+
+    // This function didn't do anything; use the built-in conversion.
+    return AVIF_RESULT_NOT_IMPLEMENTED;
+}
+
+// Returns whether libyuv was compiled with LIBYUV_BIT_EXACT defined or not.
+static avifBool avifIsLibyuvBitExact()
```

Comment:
Please ask Frank Barchard (the author of libyuv) to review this function. He may be able to suggest a better way to detect if libyuv was compiled with LIBYUV_BIT_EXACT defined.

Reply:
Done at http://crbug/libyuv/936#c9

Reply:
Shouldnt this compare ARGBToI420 results?  The difference with LIBYUV_BIT_EXACT relates to subsampling.

Reply:
As mentioned at http://crbug/libyuv/936#c10, `avifIsLibyuvBitExact()` is not a very sustainable solution. As there is no equivalent alternative in libyuv, I suggest we drop any `LIBYUV_BIT_EXACT` detection.

There is nothing to change for full range because it gives more accurate results without `LIBYUV_BIT_EXACT`, `LIBYUV_BIT_EXACT` is disabled by default in libyuv, and it is unlikely to be enabled by accident by the user.

However for limited range, here are the stats without `LIBYUV_BIT_EXACT`:
- RGB to YUV limited range (without LIBYUV_BIT_EXACT):
    Overall average drift increased by 0.26 on average, 0.32 max
    Quality loss is increased by 46% on average, 103% max
    Sample value difference is increased by 1.44 in average, up to 2
    PSNR drops by 2.2dB on average, 4.4dB max

It is significantly worse than the ["RGB to YUV limited range (with LIBYUV_BIT_EXACT)" quality loss from my original post](https://github.com/AOMediaCodec/libavif/pull/989#issue-1301079230) (results obtained with an Intel CPU). On the other hand, there is no visually detectable loss, it only impacts the objective metrics at high quality, so it might be fine. On top of that, limited range is already bringing a significant quality drop, so users of this setting should not be that concerned by this RGB-to-YUV conversion loss.

Thus I suggest we keep the libyuv calls in this PR as is but we do not check for `avifIsLibyuvBitExact()`. This will make it easier to improve libyuv accuracy in the future, if necessary.

Reply:
If there is an accuracy issue with libyuv RGB subsampling, I'd be interested to fix it.
libyuv takes the average of the 4 RGB pixels, producing a single RGB pixel.  And then proceeds to convert to UV.
Is this ok, or should the averaging be moved to after the conversion?
Do we know where 'drift' is coming from?


LIBYUV_BIT_EXACT makes intel use the arm method.  Instead of average of 4 pixels, it uses the sum of 4 pixels without the divide by 4.  Then the matrix has adjusted coefficients.  This hurts quality for some color spaces.  If this is the problem, I'd suggest making ARM / BIT EXACT use the intel method.
It just adds a shift, hurting performance on ARM by 2 instructions.

exact flag will likely go away... it makes sense for ARM and Intel to match.
But if flags need to remembered, a function could save all the compile flags into bits of an int so they can be queried or changed at runtime.

Reply:
Frank: If we ignore the clipping and rounding operations, the RGB-to-UV conversion and the four-pixel averaging operation are both linear, so we should be able to interchange them and the results should only differ due to clipping and rounding. Of course, we would prefer if the averaging is moved to after the conversion to eliminate this source of inaccuracy.